tests/conftest.py

```diff
@@ -18,5 +18,6 @@ def event_loop() -> typing.Generator[asyncio.AbstractEventLoop, None, None]:
 
 
 @pytest.fixture(autouse=True, scope="session")
-async def test_database() -> None:
+async def test_database() -> typing.AsyncGenerator:
+    yield
     await client.drop_database("test_db")
```

Comment:
Shouldn't we drop it both at the beginning and end of all tests? I mean if the database was edited manually during before the tests, that would fail the tests.

Reply:
Ideally we should be mocking the db as well instead of touching mongo.
You can look into https://github.com/mongomock/mongomock.

Reply:
In fact I strongly believe we should mock the db. There are many cases where we can end up changing the state of the db
- If someone stops the tests before finishing
- Tests fail for some reason

Reply:
TBH, I wouldn't like to mock database in the ODM package testing.
Correct me if I'm wrong, you could mock the database if you're building on top of ODM/ORM like a web application, but not when you're writing the ODM itself so I'd say the safest approach would be to test on the db itself.

Reply:
Hmm… that’s a fair point. In that case let’s cleanup before and afterwards.